### DISCUSSION TITLE
Any way to run Amethyst through a programmatic API?

### DISCUSSION THREAD

**Discussion by nikfp** [ianyh/Amethyst#1518] [Q&A]
Hello, I have a question about configuration: Does Amethyst expose a programmatic API that I can hook into, with other tools like SKHD or Hammerspoon? My goal is to automate how Amethyst runs in concert with some other tools. 

Edit: This could also work with CLI commands as both of the tools mentioned above should allow executing shell scripts.

**Comment by ianyh** [comment]
Unfortunately, no. It's just not really how the app was architected. I don't think I personally have plans to implement such functionality.

**Comment by nikfp** [comment]
fair enough, thanks for the response, and love the app as is!
